src/NServiceBus.AcceptanceTesting/ContextAppender.cs

```diff
@@ -7,150 +7,130 @@
-            RecordLog(fullMessage, "info");
         }
 
         public void InfoFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "info");
+            WriteToTrace(fullMessage, LogLevel.Info);
         }
 
         public void Warn(string message)
         {
-            Trace.WriteLine(message);
-            RecordLog(message, "warn");
+            WriteToTrace(message, LogLevel.Warn);
         }
 
         public void Warn(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            WriteToTrace(fullMessage, LogLevel.Warn);
             AppendException(exception);
-            RecordLog(fullMessage, "warn");
         }
 
         public void WarnFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "warn");
+            WriteToTrace(fullMessage, LogLevel.Warn);
         }
 
         public void Error(string message)
         {
-            Trace.WriteLine(message);
-
-            RecordLog(message, "error");
+            WriteToTrace(message, LogLevel.Error);
         }
 
         public void Error(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            WriteToTrace(fullMessage, LogLevel.Error);
             AppendException(exception);
-            RecordLog(fullMessage, "error");
         }
 
         public void ErrorFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "error");
+            WriteToTrace(fullMessage, LogLevel.Error);
         }
 
         public void Fatal(string message)
         {
-            Trace.WriteLine(message);
-
-            RecordLog(message, "fatal");
+            WriteToTrace(message, LogLevel.Fatal);
         }
 
         public void Fatal(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            WriteToTrace(fullMessage, LogLevel.Fatal);
             AppendException(exception);
-            RecordLog(fullMessage, "fatal");
         }
 
         public void FatalFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "fatal");
+            WriteToTrace(fullMessage, LogLevel.Fatal);
         }
+
+        void WriteToTrace(string message, LogLevel messageSeverity)
+        {
+            if (level <= messageSeverity)
```

Comment:
if we do this check for messages, I think we need to do it for the exceptions too.